akka/src/main/scala/cakesolutions/kafka/akka/KafkaProducerActor.scala

```diff
@@ -142,11 +234,25 @@ private class KafkaProducerActor[K, V](
     }
   }
 
-  private def logSendError(err: Throwable): Unit =
-    log.error(err, "Failed to send message to Kafka!")
+  private def logSendError(err: Throwable): Unit = log.error(err, "Failed to send message to Kafka!")
 
   private def sendMany(records: Records) = Future.sequence(records.map(send))
 
+  private def sendManyTransactional(records: Records): Future[Iterable[RecordMetadata]] =
+    Future
+      .fromTry(Try(producer.beginTransaction()))
+      .flatMap(_ =>
+        (for {
+          results <- sendMany(records)
+          _ = Future.fromTry(Try(producer.commitTransaction()))
+        } yield results).recoverWith { case sendThrowable =>
```

Comment:
Nothing here the the transaction may not have started in the first place! i.e. if `beginTransaction` failed.

Reply:
it shouldn't get here is begin fails.  thats why the result of beginTransaction is flatMapped here.